application.go

```diff
@@ -234,7 +231,7 @@ func (a *Application) Run() error {
 	a.window.SetUserPointer(unsafe.Pointer(&flutterEnginePointer))
 
 	// Start the engine
-	result := a.engine.Run(unsafe.Pointer(&flutterEnginePointer), a.config.vmArguments)
+	result := a.engine.Run(unsafe.Pointer(&flutterEnginePointer), a.config.vmArguments, IsAot == "true")
```

Comment:
I don't think we need to have another parameter.
The embedder.h has a special method that we can query `FlutterEngineRunsAOTCompiledDartCode`.

https://github.com/flutter/engine/blob/975c9c8607fa09372dc27f5bcfc7496b6d194b79/shell/platform/embedder/embedder.h#L1598-L1616

Reply:
I haven't looked at how hover set-up aot, but using `FlutterEngineRunsAOTCompiledDartCode` it might be simpler.

Reply:
I used this code:
```go
isAOT = *(*bool)(unsafe.Pointer(C.FlutterEngineRunsAOTCompiledDartCode))
```
but it always returns true.
Because of [this](https://github.com/flutter/engine/blob/1663ac9e0d7451e026b70ec5fb15133cb397fdfb/shell/platform/embedder/embedder.h#L1463-L1464) we can't use it I think, because go-flutter is configured for both JIT and AOT. Looks like we have to use the current approach even if we don't like it.

Reply:
Humm, interesting. Either the documentation is wrong or there is an issue on the engine side. Looking at the doc this is really what we need to distinguish between a JIT and AOT.
We could have a generated binary that work for both AOT and JIT linked engine..

Reply:
I'm not 100% sure 
`isAOT = *(*bool)(unsafe.Pointer(C.FlutterEngineRunsAOTCompiledDartCode))`
is correct. 
Let me take a look!

Reply:
`C.FlutterEngineRunsAOTCompiledDartCode()` works! :)